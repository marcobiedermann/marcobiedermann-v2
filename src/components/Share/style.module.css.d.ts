/* eslint-disable import/prefer-default-export */

export const share: string;
