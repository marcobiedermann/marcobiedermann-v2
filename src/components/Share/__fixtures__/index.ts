const share = {
  url: 'https://example.com',
  via: 'twitter',
};

export default share;
