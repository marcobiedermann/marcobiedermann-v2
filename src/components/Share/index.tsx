import classNames from 'classnames';
import React from 'react';
import { FacebookShareButton, TwitterShareButton } from 'react-share';
import facebookIcon from '../../images/icons/facebook.svg';
import twitterIcon from '../../images/icons/twitter.svg';
import styles from './style.module.css';

export interface ShareProps {
  className?: string;
  url: string;
  via: string;
}

const Share: React.FC<ShareProps> = (props) => {
  const { className, url, via, ...otherProps } = props;

  return (
    <ul className={classNames(className, styles.share)} {...otherProps}>
      <li>
        <FacebookShareButton url={url}>
          <img src={facebookIcon} alt="Facebook" width="32" height="32" />
        </FacebookShareButton>
      </li>
      <li>
        <TwitterShareButton url={url} via={via}>
          <img src={twitterIcon} alt="Twitter" width="32" height="32" />
        </TwitterShareButton>
      </li>
    </ul>
  );
};

export default Share;
