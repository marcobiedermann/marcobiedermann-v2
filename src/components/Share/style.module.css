.share {
  display: grid;
  gap: 0.5em;
  grid-auto-flow: column;
  justify-content: center;
  list-style: none;
  margin: 1.5em 0;
  padding: 0;
}

.SocialMediaShareButton {
  cursor: pointer;
}
