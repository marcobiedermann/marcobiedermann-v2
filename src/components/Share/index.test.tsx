/* eslint-disable import/no-extraneous-dependencies */

import React from 'react';
import renderer from 'react-test-renderer';
import Share from '.';
import share from './__fixtures__';

describe('Share component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Share {...share} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
