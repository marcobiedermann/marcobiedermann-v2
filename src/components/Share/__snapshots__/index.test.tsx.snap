// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Share component should render correctly 1`] = `
<ul
  className="share"
>
  <li>
    <button
      aria-label="facebook"
      className="react-share__ShareButton"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "border": "none",
          "color": "inherit",
          "cursor": "pointer",
          "font": "inherit",
          "padding": 0,
        }
      }
    >
      <img
        alt="Facebook"
        height="32"
        src="test-file-stub"
        width="32"
      />
    </button>
  </li>
  <li>
    <button
      aria-label="twitter"
      className="react-share__ShareButton"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "border": "none",
          "color": "inherit",
          "cursor": "pointer",
          "font": "inherit",
          "padding": 0,
        }
      }
      via="twitter"
    >
      <img
        alt="Twitter"
        height="32"
        src="test-file-stub"
        width="32"
      />
    </button>
  </li>
</ul>
`;
